### PR TITLE
feat: module3 hw

### PR DESCRIPTION
**Pull request by AlexanderSirash**

### CODE REVIEW

**Comment by AlexanderSirash** [comment]
https://www.getpostman.com/collections/a14a43aba1168894ea35

File: module_3/src/service/User.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import statusCodes from '../../statusCodes.js';
+
+class User extends API {
+  constructor() {
+    super();
+    this.users = [];
+
+  }
+
+  findUser(id) {
```
Comment (boale) [inline module_3/src/service/User.js]: `findById` or `findOne`
`User` part is not necessary as it is already scoped by `User` class

File: module_3/src/service/User.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  getUser(req, res) {
+    res.status(statusCodes.ok).json(this.findUser(req.params.id));
+  }
+
+  addUser(req, res) {
+    const user = super.filterModel(userModel, req.body);
+    const id = uuidv4();
+    this.users.push({ id, ...user, isDeleted: false });
+    res.status(statusCodes.ok).json(id);
+  }
```
Comment (boale) [inline module_3/src/service/User.js]: UserService must not be coupled with `req` or `res` objects. It must be responsible only for managing user data.

File: module_3/src/service/User.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    const filteredBySubstringUsers = sortedUsers.filter(user => user.login.includes(loginSubstring));
+    const requestedAmountOfUsers = filteredBySubstringUsers.slice(sliceStartPosition, limit);
+    res.status(statusCodes.ok).json(requestedAmountOfUsers);
+  }
+
+  removeUser(req, res) {
+    const user = this.findUser(req.params.id);
+    const updatedUser = { ...user, isDeleted: true };
+    this.updateUsers(updatedUser);
+    res.status(statusCodes.ok).json('success');
```
Comment (boale) [inline module_3/src/service/User.js]: Sending requests is a responsibility of route handler

File: module_3/src/service/User.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import API from './Api.js';
+import { userModel } from '../model/user.js';
+import { v4 as uuidv4 } from 'uuid';
+import statusCodes from '../../statusCodes.js';
+
+class User extends API {
```
Comment (boale) [inline module_3/src/service/User.js]: User must split into two services - `UserService` or `UserRepository` - which will manage user data
second one must be a `UserApiService` which will handle API requests.
S - Single Responsibility 😄

Comment (AlexanderSirash) [reply module_3/src/service/User.js]: hi, thx for the comments, i updated the code regarding the previous ones, but can't catch what did u mean in this one?  can u rephrase it, or give an example?
For this moment API created as a place to save additional helpers functions. I renamed it and remove extending it by UserService.

**Pull request merged**